pkg/sql/codegen/analyzer/codegen.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2019 The SQLFlow Authors. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.o
+
+package analyzer
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"sqlflow.org/sqlflow/pkg/sql/codegen"
+)
+
+const (
+	shapSummaryAttributes = "shap_summary"
```

Comment:
Please switch to `attribute.Dictionary` in the upcoming PRs. Example:

https://github.com/sql-machine-learning/sqlflow/blob/eb19db58b36640b58cca0e6b979be0dff7101e4a/pkg/sql/codegen/xgboost/codegen.go#L37-L47